src/libraries/System.Net.Quic/src/System/Net/Quic/QuicStream.cs

```diff
@@ -349,7 +349,7 @@ public override ValueTask WriteAsync(ReadOnlyMemory<byte> buffer, CancellationTo
     /// <param name="buffer">The region of memory to write data from.</param>
     /// <param name="cancellationToken">The token to monitor for cancellation requests. The default value is <see cref="CancellationToken.None"/>.</param>
     /// <param name="completeWrites">Notifies the peer about gracefully closing the write side, i.e.: sends FIN flag with the data.</param>
-    public async ValueTask WriteAsync(ReadOnlyMemory<byte> buffer, bool completeWrites, CancellationToken cancellationToken = default)
+    public ValueTask WriteAsync(ReadOnlyMemory<byte> buffer, bool completeWrites, CancellationToken cancellationToken = default)
```

Comment:
This is a public API where the expectation is that exceptions other than arg validation escape asynchronously rather than synchronously... without additional changes, this breaks that assumption.

Reply:
I am not sure I follow, in which scenario this code behaves differently?

Reply:
e.g. if !_canWrite, your PR will cause an InvalidOperationException to be thrown synchronously out of the method. With what's in main, if !_canWrite, the method will instead return a Task in the faulted state and storing that same InvalidOperationException. If a consumer directly awaits the returned task, they won't observe a functional difference (other than some diagnostics like stack trace), but if they instead do something like `WriteAsync(...).AsTask().ContinueWith(...)`, it's the difference between their continuation running and not running.

Reply:
I see, so replacing throw statements with return statements with exceptions wrapped in Value tasks should work, right? (And maybe try-catch in case something else can throw)